src/main/java/mate/academy/spring/service/impl/CustomUserDetailsService.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.spring.service.impl;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User userFromDb = userService.findByEmail(username)
+                .orElseThrow(() -> new UsernameNotFoundException("User not found"));
+        UserBuilder builder =
+                org.springframework.security.core.userdetails.User.withUsername(username);
```

Comment:
```suggestion
                withUsername(username);
```
Потрібно винести в Import частину назви (шлях)